src/components/pageControl/index.js

```diff
@@ -72,38 +93,113 @@ export default class PageControl extends BaseComponent {
     enlargeActive: false
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      numOfPagesShown: Math.min(MAX_SHOWN_PAGES, props.numOfPages),
+      largeIndicatorsOffset: 0,
+      pagesOffset: 0,
+      prevPage: undefined
+    };
+
+    if (this.showLimitedVersion(props)) {
+      const {smallSize, mediumSize, size} = props;
+      if (mediumSize >= size || smallSize >= mediumSize) {
```

Comment:
I see you have a lot of range comparisons in the code. 
Consider using lodash `_.inRange`

Reply:
10x, but it does not seem to be relevant here: _.inRange is not-inclusive